applications/FluidDynamicsApplication/custom_utilities/estimate_dt_utilities.h

```diff
@@ -0,0 +1,275 @@
+            }
+        }
+
+        // Obtain the maximum projected element size (compare thread results)
+        double Max = 0.0;
+        for (int k = 0; k < NumThreads; ++k)
+            if (Max < MaxProj[k]) Max = MaxProj[k];
+
+        // Dt to obtain user-defined CFL
+        double dt = mCFL / Max;
+        if(dt > mDtMax)
+            dt = mDtMax;
+
+        // Perform MPI sync if needed
+        double global_dt = dt;
+        mrModelPart.GetCommunicator().MinAll(global_dt);
+        dt = global_dt;
+
+        return dt;
+
+        KRATOS_CATCH("")
+    }
+
+    /// Calculate each element's CFL for a given time step value.
+    /**
+     * This function is mainly intended for test and debug purposes, but can
+     * be sometimes useful to detect where a mesh is inadequate. CFL values
+     * are stored as the elemental value of DIVPROJ, so be careful with elements
+     * that use it, such as Fluid and VMS (in particular, avoid printing both
+     * nodal and elemental values of the variable in GiD)
+     * @param Dt The time step used by the fluid solver
+     */
+    void CalculateLocalCFL(const double Dt)
+    {
+        KRATOS_TRY;
+
+        int NumThreads = OpenMPUtils::GetNumThreads();
+        OpenMPUtils::PartitionVector ElementPartition;
+        OpenMPUtils::DivideInPartitions(mrModelPart.NumberOfElements(),NumThreads,ElementPartition);
+
+        #pragma omp parallel
+        {
+            int k = OpenMPUtils::ThisThread();
+            ModelPart::ElementIterator ElemBegin = mrModelPart.ElementsBegin() + ElementPartition[k];
+            ModelPart::ElementIterator ElemEnd = mrModelPart.ElementsBegin() + ElementPartition[k+1];
+
+            double Area;
+            array_1d<double, TNumNodes> N;
+            boost::numeric::ublas::bounded_matrix<double, TNumNodes, TDim> DN_DX;
+
+            for( ModelPart::ElementIterator itElem = ElemBegin; itElem != ElemEnd; ++itElem)
+            {
+                // Get the element's geometric parameters
+                Geometry< Node<3> >& rGeom = itElem->GetGeometry();
+                GeometryUtils::CalculateGeometryData(rGeom, DN_DX, N, Area);
+
+                // Elemental Velocity
+                array_1d<double,3> ElementVel = N[0]*itElem->GetGeometry()[0].FastGetSolutionStepValue(VELOCITY);
+                for (unsigned int i = 1; i < TNumNodes; ++i)
+                    ElementVel += N[i]*rGeom[i].FastGetSolutionStepValue(VELOCITY);
+
+                // Maximum element size along the direction of velocity
+                double ElemProj = 0.0;
+                for (unsigned int i = 0; i < TNumNodes; ++i)
+                {
+                    double Proj = 0.0;
+                    for (unsigned int d = 0; d < TDim; ++d)
+                        Proj += ElementVel[d]*DN_DX(i,d);
+                    Proj = fabs(Proj);
+                    if (Proj > ElemProj) ElemProj = Proj;
+                }
+                itElem->SetValue(DIVPROJ,ElemProj*Dt);
```

Comment:
I think it is good if we create it. It is not a problem if it is rarely used, since unused variables are 'free'. Might I suggest CFL_NUMBER with an underscore as the name?